bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tmva/_gnn.py

```diff
@@ -148,18 +176,20 @@ def add_update_function(gin, component_model, graph_type, function_target):
         function_target: Target for the function to update either of nodes, edges or globals
 
     """
-    if (component_model.name == 'mlp'):
+    if (type(component_model).__name__ == 'MLP'):
```

Comment:
Can we keep the previous approach here? Using the `__name__` attribute which is supposedly kept private might be risky if changes are introduced in the future versions.

Reply:
We cannot use the "name" attribute, because this is arbitrary given by the user, and only in case when is not specified is "mlp".  __name__  is the name of the class type, it is true it could change but is not very likely, in that case probably other differences will be present.

Reply:
oh okay, makes sense then.